receiver/zipkinreceiver/proto_parse_test.go

```diff
@@ -190,5 +191,7 @@ func TestConvertSpansToTraceSpans_protobuf(t *testing.T) {
 		},
 	}
 
-	assert.Equal(t, want, reqs)
```

Comment:
`assert.Equal` invokes `reflect.DeepEqual` internally, which is incompatible with the new protobuf, see https://developers.google.com/protocol-buffers/docs/reference/go/faq#deepequal 

Replaced with `go-cmp` as recommended there.

Reply:
A bit inconsistent, sometimes using proto.Equal sometimes using cmp.Diff. can we set on using proto.Equal?

Reply:
I've switched to `proto.Equal` when comparing proto messages directly, and to `cmp.Diff` when comparing custom data structures with proto messages inside.

e.g. here we are comparing `[]consumerdata.TraceData` slices. TraceData is not a proto message, but its fields are: https://github.com/open-telemetry/opentelemetry-collector/blob/2c3bca6dca6d12447a00f394c6f16183a809e7d1/consumer/consumerdata/consumerdata.go#L32-L38

So to use `proto.Equal` instead of `cmp.Diff` here I'd have to write a for loop comparing individual fields, and a nested loop over spans.